### PR TITLE
!Feat: Dorq task logs command. get_logs() function accepts task invocation ID instead of task run ID

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

User needs to have CLI command to get logs for individual task

# This PR's solution

Create dorq task logs command.
Also refactor API function get_logs as it was inconsistent with other commands

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

File: src/orquestra/sdk/_base/_api.py [hunk 1]
```diff
@@ -607,7 +607,7 @@ def get_artifacts(
 
     def get_logs(
         self,
-        tasks: t.Union[TaskRunId, t.List[TaskRunId]],
+        tasks: t.Union[TaskInvocationId, t.List[TaskInvocationId]],
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_api.py:610]: **non-blocking, thought**: Hot take: `tasks` and `only_available` arguments in the `sdk.WorkflowRun` methods have very little value now that we have `sdk.WorkflowRun.get_tasks()`. I'd simply get rid of them.

**Review by alexjuda — CHANGES_REQUESTED** [review]

File: src/orquestra/sdk/_base/_api.py (hunk 1 above)
Comment (alexjuda) [reply src/orquestra/sdk/_base/_api.py:610]: FYI: I've done just that for `.get_artifacts()` in https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/23

File: tests/sdk/v2/test_api.py [hunk 2]
```diff
@@ -151,10 +154,28 @@ def mock_runtime():
         }
         # Get logs, the runtime interface returns invocation IDs
         runtime.get_full_logs.return_value = {
-            "task_invocation1": ["woohoo!\n"],
-            "task_invocation2": ["another\n", "line\n"],
-            "task_invocation3": ["hello\n", "a log\n"],
+            "task_run1": ["woohoo!\n"],
+            "task_run2": ["another\n", "line\n"],
+            "task_run3": ["hello\n", "a log\n"],
```
[the remark this answers is not in the document]
Comment (SebastianMorawiec) [reply tests/sdk/v2/test_api.py:159]: Ah yes, Ive looped around that what should be here and eventually got the wrong thing :D

File: tests/cli/dorq/test_repos.py [hunk 3]
```diff
@@ -569,6 +569,65 @@ def test_passing_errors(monkeypatch, exc):
                     # When
                     _ = repo.get_wf_logs(wf_run_id, config)
 
+        class TestGetTaskLogs:
+            @staticmethod
+            def test_passing_values(monkeypatch, mock_by_id, mock_wf_run):
```
Comment (alexjuda) [inline tests/cli/dorq/test_repos.py:574]:
```suggestion
            def test_passing_values(mock_by_id, mock_wf_run):
```

File: tests/cli/dorq/test_repos.py [hunk 4]
```diff
@@ -569,6 +569,65 @@ def test_passing_errors(monkeypatch, exc):
...
+
+                # Then
+                assert repo.get_task_logs(wf_run_id, task_run_id, config) == logs
+
+            @staticmethod
+            @pytest.mark.parametrize(
+                "exc",
+                [exceptions.WorkflowRunNotStarted(), exceptions.TaskRunNotFound()],
+            )
+            def test_passing_errors(monkeypatch, mock_by_id, mock_wf_run, exc):
```
Comment (alexjuda) [inline tests/cli/dorq/test_repos.py:594]:
```suggestion
            def test_passing_errors(mock_by_id, mock_wf_run, exc):
```

File: tests/cli/dorq/test_repos.py [hunk 5]
```diff
@@ -569,6 +569,65 @@ def test_passing_errors(monkeypatch, exc):
...
+                with pytest.raises(type(exc)):
+                    # When
+                    _ = repo.get_task_logs(wf_run_id, task_run_id, config)
+
+            @staticmethod
+            @pytest.mark.parametrize(
+                "exc",
+                [exceptions.NotFoundError(), exceptions.ConfigNameNotFoundError()],
+            )
+            def test_wf_not_found(monkeypatch, mock_by_id, exc):
```
Comment (alexjuda) [inline tests/cli/dorq/test_repos.py:615]:
```suggestion
            def test_wf_not_found(mock_by_id, exc):
```

**Review by alexjuda — APPROVED** [review]

**Pull request merged**